tools/SetupFlow/DevHome.SetupFlow.ElevatedComponent/ElevatedComponentOperation.cs

```diff
@@ -18,14 +18,32 @@ public sealed class ElevatedComponentOperation : IElevatedComponentOperation
     /// Tasks arguments are passed to the elevated process as input at launch-time.
     /// </summary>
     /// <remarks>
-    /// This object is used to ensures that the operations performed at runtime by the
+    /// This object is used to ensure that the operations performed at runtime by the
     /// caller process were pre-approved.
     /// </remarks>
     private readonly TasksArguments _tasksArguments;
 
+    /// <summary>
+    /// Dictionary of operations state by task arguments.
+    /// </summary>
+    private readonly Dictionary<ITaskArguments, OperationState> _operationsState = new ();
```

Comment:
Should we add `GetHashCode()` to the task arguments? Otherwise, it would be depending on the keys being the same object to find them here.

Reply:
The dictionary key comparison should be the exact object reference, we only maintain one copy in memory for each task arguments

Reply:
Is there a way to ensure that we only maintain one copy?

Reply:
The tasks arguments are created in the constructor. For each operation performed, we first gets the corresponding reference to the task argument object in memory and uses that as a key to query the dictionary. If an object that was not created in the constructor was used for an operation, the latter will fail because the process does not recognize that operation (aka not approved)